src/desktop/src/ui/views/account/Tools.js

```diff
@@ -51,13 +51,23 @@ class Addresses extends PureComponent {
         t: PropTypes.func.isRequired,
     };
 
-    componentWillReceiveProps(newProps) {
-        const { wallet, ui } = newProps;
+    componentDidUpdate(prevProps) {
+        const { wallet, ui } = this.props;
+
+        if (
+            prevProps.isTransitioning === ui.isTransitioning &&
+            prevProps.isAttachingToTangle === ui.isAttachingToTangle &&
+            prevProps.balanceCheckFlag === wallet.balanceCheckFlag &&
+            prevProps.ui.isSyncing === ui.isSyncing
+        ) {
+            return false;
```

Comment:
Is there a reason for returning a boolean in `componentDidUpdate`?

Reply:
No, will update